connector/spanmetricsconnector/connector.go

```diff
@@ -44,8 +44,66 @@ const (
 	metricNameEvents   = "events"
 
 	defaultUnit = metrics.Milliseconds
+
+	// https://github.com/open-telemetry/opentelemetry-go/blob/3ae002c3caf3e44387f0554dfcbbde2c5aab7909/sdk/metric/internal/aggregate/limit.go#L11C36-L11C50
+	overflowKey = "otel.metric.overflow"
 )
 
+type cardinalityTracker struct {
+	limit          int
+	seenDimensions map[string]map[metrics.Key]struct{} // metricName -> key
```

Comment:
Sorry for the delay. In `spanmetricsconnector/internal/metrics` We have already cached all unique keys
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/8d85c2f3e42236d52cd4d100f1ae38067433f953/connector/spanmetricsconnector/internal/metrics/metrics.go#L262-L265

we can move the limit logic to `GetOrCreate` method. It's can avoid add a new cache map.